### PR TITLE
Add pre-install hook for natss

### PR DESCRIPTION
**Pull request by radufa**
**Description**

Changes proposed in this pull request:

- Adds a pre-install Helm 3 hook for waiting till the "knative-eventing" namespace was created by another parallel deployment. The timeout is 4 mins, under the default timeout of Helm 3 installer (5 mins)

**Related issue(s)**

See also #8527

### CODE REVIEW

File: resources/nats-streaming/check-namespace.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+until [ $counter -gt $seconds ]; do
+  timestamp=`date`
+  if kubectl get ns | grep knative-eventing; then
+    echo "[$timestamp] knative-eventing found"
+    break
+  else
+    echo "[$timestamp] knative-eventing not found"
+  fi
+  sleep 1
+  ((counter++))
```
Comment (k15r) [inline resources/nats-streaming/check-namespace.sh:13]: is it really worth checking that often?

File: resources/nats-streaming/templates/pre-install-hook.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Namespace }}-ns-check
```
Comment (k15r) [inline resources/nats-streaming/templates/pre-install-hook.yaml:4]: why not .Release.Name??  (for all of them)

File: resources/nats-streaming/templates/pre-install-hook.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Release.Namespace }}-ns-check
+  annotations:
+    helm.sh/hook: pre-install
```
Comment (k15r) [inline resources/nats-streaming/templates/pre-install-hook.yaml:9]: do we also need pre-upgrade?

File: resources/nats-streaming/check-namespace.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+seconds=240    #seconds to wait till "knative-eventing namespace" will be created by another Helm 3 installer
+counter=0
+until [ $counter -gt $seconds ]; do
+  timestamp=`date`
+  if kubectl get ns | grep knative-eventing; then
```
Comment (k15r) [inline resources/nats-streaming/check-namespace.sh:6]:
```suggestion
  if kubectl get ns knative-eventing; then
```

File: resources/nats-streaming/check-namespace.sh (hunk 1 above)
Comment (radufa) [reply resources/nats-streaming/check-namespace.sh:13]: Change to wait for 5s before checking again

File: resources/nats-streaming/templates/pre-install-hook.yaml (hunk 2 above)
Comment (radufa) [reply resources/nats-streaming/templates/pre-install-hook.yaml]: Done

File: resources/nats-streaming/check-namespace.sh (hunk 4 above)
Comment (radufa) [reply resources/nats-streaming/check-namespace.sh]: done

File: resources/nats-streaming/templates/pre-install-hook.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+metadata:
+  name: pre-install-job
+  annotations:
+    helm.sh/hook: "pre-install"
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    sidecar.istio.io/inject: "false"
+spec:
+  backoffLimit: 1
+  template:
+    metadata:
```
Comment (nachtmaar) [inline resources/nats-streaming/templates/pre-install-hook.yaml:55]: (nit) can we have a label for the container executing the job? For uninstallation we have [this one](https://github.com/kyma-project/kyma/blob/master/resources/knative-eventing/charts/knative-eventing/templates/uninstall-eventing.yaml#L829=

File: resources/nats-streaming/templates/pre-install-hook.yaml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: {{ .Release.Name }}-ns-check
+      restartPolicy: Never
+      terminationGracePeriodSeconds: 0
+      containers:
+        - name: {{ .Release.Name }}-pre-install
+          image: eu.gcr.io/kyma-project/test-infra/alpine-kubectl:v20200507-070ff576
```
[the remark this answers is not in the document]
Comment (radufa) [reply resources/nats-streaming/templates/pre-install-hook.yaml]: Done

File: resources/nats-streaming/templates/pre-install-hook.yaml (hunk 5 above)
Comment (radufa) [reply resources/nats-streaming/templates/pre-install-hook.yaml:55]: Done

**Pull request merged**